crates/core/src/transform.rs

```diff
@@ -277,48 +277,122 @@ impl TransformSystem {
                         }
                     }),
                     // FBX
-                    query((fbx_complex_transform(), local_to_parent()))
-                        .optional_changed(translation())
-                        .optional_changed(fbx_rotation_offset())
-                        .optional_changed(fbx_rotation_pivot())
-                        .optional_changed(fbx_pre_rotation())
-                        .optional_changed(rotation())
-                        .optional_changed(fbx_post_rotation())
-                        .optional_changed(fbx_rotation_pivot())
-                        .optional_changed(fbx_scaling_offset())
-                        .optional_changed(fbx_scaling_pivot())
-                        .optional_changed(scale())
-                        .optional_changed(fbx_scaling_pivot())
-                        .to_system(|q, world, qs, _| {
-                            // See: https://help.autodesk.com/view/FBX/2017/ENU/?guid=__files_GUID_10CDD63C_79C1_4F2D_BB28_AD2BE65A02ED_htm
-                            // and: https://github.com/assimp/assimp/blob/add7f1355e96c6ff0df0ba3cec084f25332d154e/code/AssetLib/FBX/FBXConverter.cpp#L687
-                            for (id, _) in q.collect_cloned(world, qs) {
-                                world
-                                    .set(id, local_to_parent(), get_fbx_transform(world, id))
-                                    .unwrap();
-                            }
-                        }),
-                    query((fbx_complex_transform(), local_to_world()))
-                        .optional_changed(translation())
-                        .optional_changed(fbx_rotation_offset())
-                        .optional_changed(fbx_rotation_pivot())
-                        .optional_changed(fbx_pre_rotation())
-                        .optional_changed(rotation())
-                        .optional_changed(fbx_post_rotation())
-                        .optional_changed(fbx_rotation_pivot())
-                        .optional_changed(fbx_scaling_offset())
-                        .optional_changed(fbx_scaling_pivot())
-                        .optional_changed(scale())
-                        .optional_changed(fbx_scaling_pivot())
-                        .to_system(|q, world, qs, _| {
-                            // See: https://help.autodesk.com/view/FBX/2017/ENU/?guid=__files_GUID_10CDD63C_79C1_4F2D_BB28_AD2BE65A02ED_htm
-                            // and: https://github.com/assimp/assimp/blob/add7f1355e96c6ff0df0ba3cec084f25332d154e/code/AssetLib/FBX/FBXConverter.cpp#L687
-                            for (id, _) in q.collect_cloned(world, qs) {
-                                world
-                                    .set(id, local_to_world(), get_fbx_transform(world, id))
-                                    .unwrap();
-                            }
-                        }),
+                    query_mut(
```

Comment:
I'm not sure that they will always have _all_ of these components, which was why it was implemented as it was.

Reply:
Oh I see, you're always setting them now. Yeah I guess that's fine

Reply:
Yeah, I did an unsound optional variant that got a modest ~25% increase in performance. I think in the case of transforms it should be worth to spend a little bit more on memory to utilize the hardware better.